### PR TITLE
[RAPTOR-7748] pass dir/spooler_type params to the embedded mlops monitor, release v1.9.10

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Pass dir/spooler_type params to the embedded mlops monitor. They are optional now, but will be enforced starting 8.2.X (or whatever version is released next into PyPi)

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_integration_in_a_single_container

File: custom_model_runner/datarobot_drum/drum/args_parser.py [hunk 1]
```diff
@@ -946,6 +946,8 @@ def verify_monitoring_options(options, parser_name):
                     missing_args.append(ArgumentsOptions.DR_WEBSERVER)
                 if options.api_token is None:
                     missing_args.append(ArgumentsOptions.DR_API_TOKEN)
+                if options.monitor_settings is None:
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/drum/args_parser.py:949]: Will it make sense to support default monitor settings? With that, we'll avoid compatibility issue with DataRobot.

**Review by zohar-mizrahi — CHANGES_REQUESTED** [review]

**Comment by yakov-g** [comment]
jarvis run test_integration_in_a_single_container

**Comment by yakov-g** [comment]
jenkins run test_inference_model_templates

**Comment by yakov-g** [comment]
jenkins please run test_inference_model_templates

**Comment by yakov-g** [comment]
jenkins run test_drop_in_envs

**Comment by yakov-g** [comment]
jenkins run test_drop_in_envs

**Review by gkoundry — APPROVED** [review]
LGTM Core Modeling

**Review by jlandersen — APPROVED** [review]

File: custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py [hunk 2]
```diff
@@ -53,10 +56,19 @@ def mlpiper_configure(self, params):
 
     def _init_mlops(self, params):
 
+        if not self._params["monitor_settings"]:
+            self._mlops_spool_dir = tempfile.mkdtemp()
+            monitor_settings = "spooler_type=FILESYSTEM;directory={};max_files=5;file_max_size=10485760".format(
+                self._mlops_spool_dir
+            )
+        else:
+            monitor_settings = self._params["monitor_settings"]
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:65]: I think a safer and cleaner way is:

```suggestion
        monitor_settings = self._params.get("monitor_settings")
        if not monitor_settings:
            self._mlops_spool_dir = tempfile.mkdtemp()
            monitor_settings = "spooler_type=FILESYSTEM;directory={};max_files=5;file_max_size=10485760".format(
                self._mlops_spool_dir
            )
```
(But, your call)

File: custom_model_runner/datarobot_drum/drum/args_parser.py (hunk 1 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/args_parser.py]: If monitor settings are not provided, we consider default spool channel to be filesystem

**Pull request merged**